probe-rs/src/debug/variable.rs

```diff
@@ -655,214 +699,239 @@ impl Variable {
+                children,
+                &type_name,
+                show_name,
+            ),
+        })
     }
 
     /// Calculate the memory range that contains the value of this variable.
+    ///
     /// If the location and/or byte size is not known, then return None.
-    /// Note: We don't do any validation of the memory range here, and leave it up to the caller to
-    /// validate the memory ranges before attempting to read them.
+    /// Note: We don't do any validation of the memory range here and leave it
+    /// up to the caller to validate the memory ranges before attempting to read
+    /// them.
     pub fn memory_range(&self) -> Option<Range<u64>> {
-        if let VariableLocation::Address(address) = self.memory_location {
-            if self.byte_size.is_some_and(|byte_size| byte_size.is_zero()) {
-                // This can happen for instance with empty arrays, and even though we don't need to read the
-                // array data, we need to read the array structure type.
-                Some(address..address + 4)
-            } else {
-                self.byte_size
-                    .map(|byte_size| address..(address + byte_size))
-            }
+        let VariableLocation::Address(address) = self.memory_location else {
+            return None;
+        };
+
+        self.byte_size.map(|byte_size| {
+            byte_size
+                .is_zero()
+                .then(|| address..address + 4)
+                .unwrap_or_else(|| address..(address + byte_size))
+        })
+    }
+}
+
+/// Format a pointer value
+///
+/// Formats the pointed to value and potential subsequent children as well.
+fn format_pointer_value(
+    variable_cache: &VariableCache,
+    indentation: usize,
+    first_child: Option<&Variable>,
+) -> String {
+    let line_start = line_indent_string(indentation);
+
+    let value = if let Some(first_child) = first_child {
+        first_child
+            .formatted_variable_value(variable_cache, indentation + 1, true)
+            .expect("a child. This is a bug. Please report it.")
+    } else {
+        "Unable to resolve referenced variable value".to_string()
+    };
+
+    format!("{line_start}{value}",)
+}
+
+/// Format any array like value.
+///
+/// Recursively formats all child values.
+fn format_array_value<'a>(
+    variable_cache: &VariableCache,
+    indentation: usize,
+    children: &mut (impl Iterator<Item = &'a Variable> + Clone),
+    type_name: &str,
+) -> String {
+    let line_start = line_indent_string(indentation);
+
+    // Limit arrays to 10 elements
+    const ARRAY_MAX_LENGTH: usize = 10;
+
+    let children_values = children
+        .by_ref()
+        .take(ARRAY_MAX_LENGTH)
+        .filter_map(|child| child.formatted_variable_value(variable_cache, indentation + 1, false))
+        .join(",");
+
+    let remaining = children.count();
+    let remainder = if remaining > 0 {
+        format!(",\n{line_start}\t... and {} more", remaining)
```

Comment:
Could we please avoid reintroducing `... and 1 more` into our output?

Reply:
Ah, sorry, I completely disregarded that mechanic and did not have a closer look what it does :/

Reply:
This case makes a lot of sense.  That said, I would like to caution that  some parts of the implementation (e.g.  formatting for UX) requires “opinion” , and if everyone who touches the code asserts a new opinion, we just cause churn.